relay-server/src/services/processor.rs

```diff
@@ -3366,4 +3387,158 @@ mod tests {
             "Update `MEASUREMENT_MRI_OVERHEAD` if the naming scheme changed."
         );
     }
+
+    #[tokio::test]
+    async fn test_process_metrics_bucket_metadata() {
+        let mut token = Cogs::noop().timed(ResourceId::Relay, AppFeature::Unattributed);
+        let project_key = ProjectKey::parse("a94ae32be2584e0bbd7a4cbb95971fee").unwrap();
+        let start_time = Instant::now();
+        let config = {
+            let config_json = serde_json::json!({
+                "processing": {
+                    "enabled": true,
+                    "kafka_config": [],
+                }
+            });
+            Config::from_json_value(config_json).unwrap()
+        };
+
+        let (project_cache, mut project_cache_rx) = Addr::custom();
+        let processor =
+            create_test_processor_with_custom_service(config, Some(project_cache.clone()));
+
+        let mut item = Item::new(ItemType::Statsd);
+        item.set_payload(
+            ContentType::Text,
+            "transactions/foo:3182887624:4267882815|s",
+        );
+        for (override_received_at_metadata, expected_received_at) in [
+            (true, Some(UnixTimestamp::from_instant(start_time))),
+            (false, None),
+        ] {
+            let message = ProcessMetrics {
+                items: vec![item.clone()],
+                project_key,
+                keep_metadata: true,
+                start_time,
+                sent_at: Some(Utc::now()),
+                override_received_at_metadata,
+            };
+            processor.handle_process_metrics(&mut token, message);
+
+            let value = project_cache_rx.recv().await.unwrap();
+            if let ProjectCache::MergeBuckets(merge_buckets) = value {
+                let buckets = merge_buckets.buckets();
+                assert_eq!(buckets.len(), 1);
+                assert_eq!(buckets[0].metadata.received_at, expected_received_at);
+            } else {
+                panic!();
+            }
+        }
+    }
+
+    #[tokio::test]
+    async fn test_process_batched_metrics_bucket_metadata() {
```

Comment:
Will add integration tests in a follow up PR where we are also going to write the data to kafka.